sw/device/lib/crypto/api.h

```diff
@@ -0,0 +1,1378 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_CRYPTO_API_H_
+#define OPENTITAN_SW_DEVICE_LIB_CRYPTO_API_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Enum to handle return values of the crypto API.
+ */
+typedef enum crypto_status {
+  // Status is OK; no errors.
+  kCryptoStatusOK = 0x7d3a,
+  // Invalid input arguments; Wrong length or invalid type.
+  kCryptoIncorrectInput = 0x6b7f,
+  // Inconsistencies when cross-checking results, witness, checksums.
+  kCryptoStatusInternalError = 0x4753,
+  // An asynchronous operation is still in progress.
+  kCryptoStatusAsyncIncomplete = 0xe2b7
+} crypto_status_t;
+
+/**
+ * Enum to handle return values of the verification APIs.
+ */
+typedef enum verification_status {
+  // Return value for successful verification.
+  kVerificationStatusPass = 0x2c89,
+  // Return value for unsuccessful verification.
+  kVerificationStatusFail = 0x49d3
+} verification_status_t;
+
+/**
+ * Enum for opentitan crypto modes using a key.
+ *
+ * Used to denote crypto mode for which key usage is intended.
+ */
+typedef enum key_mode {
+  // AES ECB mode.
+  kKeyModeAesEcb = 0x0101,
+  // AES CBC mode.
+  kKeyModeAesCbc = 0x0102,
+  // AES CFB mode.
+  kKeyModeAesCfb = 0x0103,
+  // AES OFB mode.
+  kKeyModeAesOfb = 0x0104,
+  // AES CTR mode.
+  kKeyModeAesCtr = 0x0105,
+  // AES GCM mode.
+  kKeyModeAesGcm = 0x0106,
+  // AES KWP mode.
+  kKeyModeAesKwp = 0x0107,
+  // HMAC-SHA-256.
+  kKeyModeHmacSha256 = 0x0201,
+  // KMAC-256.
+  kKeyModeKmac256 = 0x0202,
+  // RSA key generation.
+  kKeyModeRsaKeygen = 0x0301,
+  // RSA digital signature.
+  kKeyModeRsaSign = 0x0302,
+  // Elliptic curve key generation.
+  kKeyModeEccKeygen = 0x0401,
+  // Elliptic curve digital signature.
+  kKeyModeEccSign = 0x0402,
+  // Elliptic curve Diffie–Hellman key exchange.
+  kKeyModeEccEcdh = 0x0403
+} key_mode_t;
+
+/**
+ * Type of blinded keys.
+ *
+ * This type is not exposed in the API because it should not be used directly.
+ */
+extern struct crypto_blinded_key_t;
```

Comment:
Drop the `extern`; this should be `typedef struct crypto_blinded_key crypto_blinded_key_t;` (which looks super silly).